feat: add non standard intrinsic signature with standard_function compiler_option

parth121101 opened the pull request:
Towards: #492

parth121101 commented:
With these changes, the following is the behaviour.
```
cat a.f90            
                   
program main
    print *, besj0(673.82)
end program %     
                                                                                                                                                      
 lfortran a.f90                          
2.28310693e-02

 lfortran a.f90 --std_functions true
semantic error: Intrinsic function `besj0` is not supported with implicit interface
 --> a.f90:2:14
  |
2 |     print *, besj0(673.82)
  |              ^^^^^^^^^^^^^ 


Note: Please report unclear, confusing or incorrect messages as bugs at
https://github.com/lfortran/lfortran/issues.


lfortran a.f90 --std_functions     
--std_functions: 1 required BOOLEAN missing
Run with --help for more information.
```

Tough I have initialized std_function option with a false, I need to add a bool while using this option. I am missing out on something but unable to figure out at what point.

HarshitaKalani reviewed (commented):
Maybe you can check src/bin/lfortran.cpp file. We set compiler options to `true` or `false` according to the standards there.

certik commented:
It has to work the other way round: these functions should be disabled by default, since we want to encourage to use the standard functions.

Are you aware of some code that uses these non-standard extensions?

I would not even bother implementing these, unless they are used by some code and we want to support it.

If we want to support these extensions, then they should be disabled by default, but you can enable them with a compiler flag, say `--non-standard-functions`.

parth121101 commented:
I have added the option non_standard_function option.  Currently I am facing an issue:
```
cat a.f90           
                   
program main
    print *, besJ1(65.7)
end program %             
                                                                                                                                              
 % lfortran a.f90 --non-standard-functions

Non-standard functions are disabled
semantic error: Intrinsic function `besj1` is not supported with implicit interface
 --> a.f90:2:14
  |
2 |     print *, besJ1(65.7)
  |              ^^^^^^^^^^^ 


Note: Please report unclear, confusing or incorrect messages as bugs at
https://github.com/lfortran/lfortran/issues.


 % lfortran a.f90                 
        
Non-standard functions are disabled
semantic error: Intrinsic function `besj1` is not supported with implicit interface
 --> a.f90:2:14
  |
2 |     print *, besJ1(65.7)
  |              ^^^^^^^^^^^ 


Note: Please report unclear, confusing or incorrect messages as bugs at
https://github.com/lfortran/lfortran/issues.
```


The issue I am facing is in either of the case the non_standard_function flag remains false and it doesn't get's updated. I went on with the solution provided by @HarshitaKalani and couldn't really find any solution to it. I have added some logs for debugging purpose.

If someone can look into it and help me out it would be very grateful.

certik commented:
This is low priority. Given the increased maintenance burden on us to test these non-standard extensions, the easiest is to not support them at all.

Is this needed for some code?

If not, then I recommend to focus on the high priority intrinsics and finish those.
